src/log-store/src/kafka/producer.rs

```diff
@@ -106,6 +107,15 @@ impl ProducerClient for PartitionClient {
         records: Vec<Record>,
         compression: Compression,
     ) -> rskafka::client::error::Result<Vec<i64>> {
+        let total_size = records.iter().map(|r| r.approximate_size()).sum::<usize>();
+        let partition = self.partition().to_string();
```

Comment:
Adding a `&str` return to `partition` might be efficient due to string creation for metrics.

Reply:
`.partition()` returns an i32, it has to be formatted to a string